rastervision/runner/out_of_process_experiment_runner.py

```diff
@@ -0,0 +1,80 @@
+import os
+import click
+
+from rastervision.runner import ExperimentRunner
+from rastervision.utils.files import save_json_config
+from rastervision.cli import Verbosity
+
+
+def make_command(command_config_uri, tmp_dir=None):
+    verbosity = Verbosity.get()
+    v_flag = 'v' * max(0, verbosity - 1)
+    if v_flag:
+        v_flag = '-{}'.format(v_flag)
+    if tmp_dir is None:
+        command = 'python -m rastervision {} run_command {}'.format(
+            v_flag, command_config_uri)
+    else:
+        command = 'python -m rastervision {} run_command {} --tempdir {}'.format(
+            v_flag, command_config_uri, tmp_dir)
+    return command
+
+
+class OutOfProcessExperimentRunner(ExperimentRunner):
+    def __init__(self):
+        self.tmp_dir = None
+
+    def _run_experiment(self, command_dag):
+        """Runs all commands."""
+
+        ids_to_job = {}
+        for command_id in command_dag.get_sorted_command_ids():
+            command_def = command_dag.get_command_definition(command_id)
+            command_config = command_def.command_config
+            command_root_uri = command_config.root_uri
+            command_uri = os.path.join(command_root_uri, 'command-config.json')
+            print('Saving command configuration to {}...'.format(command_uri))
+            save_json_config(command_config.to_proto(), command_uri)
+
+            parent_job_ids = []
+            for upstream_id in command_dag.get_upstream_command_ids(
+                    command_id):
+                if upstream_id not in ids_to_job:
+                    cur_command = (command_config.command_type, command_id)
+                    u = command_dag.get_command(upstream_id)
+                    upstream_command = (u.command_type, upstream_id)
+                    raise Exception(
+                        '{} command has parent command of {}, '
+                        'but does not exist in previous batch submissions - '
+                        'topological sort on command_dag error.'.format(
+                            cur_command, upstream_command))
+                parent_job_ids.append(ids_to_job[upstream_id])
+
+            batch_run_command = make_command(command_uri, self.tmp_dir)
+            job_id = self.submit(
```

Comment:
I don't think the `OutOfProcess..` should call a `submit` method that passes a `job_queue` and `job_def` since that is too tied to Batch concepts. A better way might be to call a `run_command` method that omits the `job_queue`, `job_def` and `attempts`. (I'll admit that `run_command` isn't perfect either since it sort of implies that something will happen synchronously, when in fact it could be asynchronous in the case of Batch, so I'm ok with leaving it as `submit` if you like that better). These don't need to be passed anyway since they are attached to `self`. I don't see anyway around passing the `parent_job_ids` though.

Reply:
I am not sure that I understand what is written here, but I have turned the respective submit functions into class methods which removes the need for some parameters.